src/main/java/uk/gov/hmcts/reform/sscs/service/LetterUtils.java

```diff
@@ -197,4 +191,112 @@ public static Stream<Pair<String, YesNo>> buildOtherPartiesForReasonableAdjustme
         }
         return otherPartyReasonableAdjustmentList.stream();
     }
+
+    public static String getNameForSender(SscsCaseData sscsCaseData) {
+        if (nonNull(sscsCaseData.getOriginalSender())) {
+            final String originalSenderCode = sscsCaseData.getOriginalSender().getValue().getCode();
+            if (originalSenderCode.equalsIgnoreCase("appellant")) {
+                return sscsCaseData.getAppeal().getAppellant().getName().getFullNameNoTitle();
+            } else if (originalSenderCode.equalsIgnoreCase("representative")) {
+                return SendNotificationHelper.getRepSalutation(sscsCaseData.getAppeal().getRep(), false);
+            } else if (originalSenderCode.equalsIgnoreCase("jointParty")) {
+                return sscsCaseData.getJointParty().getName().getFullNameNoTitle();
+            } else {
+                return getOtherPartyName(sscsCaseData)
+                        .map(Name::getFullNameNoTitle).orElse("");
+            }
+        }
+        return "";
+    }
+
+    private static Optional<Representative> getRepresentativeOfOtherParty(SscsCaseData sscsCaseData) {
+        for (CcdValue<OtherParty> op : sscsCaseData.getOtherParties()) {
+            if (op.getValue().hasRepresentative() && sscsCaseData.getOriginalSender().getValue().getCode().contains(op.getValue().getRep().getId())) {
+                return Optional.of(op.getValue().getRep());
+            }
+        }
+        return Optional.empty();
+    }
+
+    public static Optional<Name> getOtherPartyName(SscsCaseData sscsCaseData) {
+        if (sscsCaseData.getOriginalSender().getValue().getCode().contains("otherPartyRep")) {
+            Optional<Representative> representative = getRepresentativeOfOtherParty(sscsCaseData);
+            return representative.isPresent() ? Optional.of(representative.get().getName())
+                    : Optional.empty();
+        }
+        return sscsCaseData.getOtherParties().stream()
+            .filter(op -> sscsCaseData.getOriginalSender().getValue().getCode().contains(op.getValue().getId()))
+            .findFirst()
+            .map(op -> op.getValue().getName());
+    }
+
+    public static String getNotificationTypeForActionFurtherEvidence(NotificationWrapper notificationWrapper, SubscriptionWithType subscriptionWithType) {
+        if (notificationWrapper.getNotificationType().getEvent().equals(EventType.ACTION_FURTHER_EVIDENCE)) {
+            if (isValidAppellantForSetAsideRequest(notificationWrapper, subscriptionWithType)
+                    || isValidAppellantRepresentativeForSetAsideRequest(notificationWrapper, subscriptionWithType)
+                    || isValidJointPartyForSetAsideRequest(notificationWrapper, subscriptionWithType)
+                    || isValidOtherParty(notificationWrapper, subscriptionWithType)) {
```

Comment:
I would suggest combining all of these into a helper method like isValidForSetAsideConfirmation

Reply:
fixed